docs/contributing/testing.md

```diff
@@ -0,0 +1,152 @@
+```
+
++ Start a web server on a local port (eg: http://localhost:8000) and use the command line interface to request a tunnel to the local server.
+
+```bash
+lt --port 8000
+```
+
++ You will receive a URL, for example https://gqgh.localtunnel.me, that you can share with anyone for as long as your local instance remains active. Any requests will be routed to your local service at the specified port.
+
+### Testing with Sauce Labs
+
+[Sauce Labs](https://saucelabs.com/beta/dashboard/builds) allows collaborators and contributors to execute automated tests against Selenium WebDriver on pre-configured browser matrix's.
+
+The cross browser testing strategy focuses on the differences between browser rendering engines. There are three different [configuration files](https://github.com/Microsoft/fast-dna/tree/master/build/testing) based on feature maturity of ongoing work.
+
+[Alpha](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.alpha.js) Immature feature development
+
+```bash
+node build/testing/sauce-labs/test-browsers.js alpha
+```
+
+[Beta](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.beta.js) Stable feature development
+
+```bash
+node build/testing/sauce-labs/test-browsers.js beta
+```
+
+[Release](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.release.js) Mature feature development
+
+```bash
+node build/testing/sauce-labs/test-browsers.js release
+```
+
+### Testing with Applitools
+
+[Applitools](https://eyes.applitools.com/) can be used to visual regression test UI changes on your own branches allow  to execute automated tests against Selenium WebDriver. Currently, ChromeDriver and FirefoxDriver are supported.
+
+Prior to beginning to make UI changes on a new branch it's recommended to run initially to create a baseline set of images.
+
+1. The first time the test is executed on a new branch, screenshots are created as 'undefined' status. 
+2. After the tests finish users should select all checkbox's and choose to save to generate the baseline image.
+3. Next perform UI code changes
+4. Rerun the first set of tests and for all snapshots that match the baseline they will turn green and show as 'passed' status. Any 'undefined' status would be additional UI changes added. Though to get into this scenario new test cases would be required in the `./run-msft-docs.js` file.  Any anomalies will be marked red as 'failed' status.
+5. Manually review all UI changes and if the changes are desired select and choose to save the new baseline. Anything, else should be changed and retested.
+
+```bash
+node build/testing/applitools/test-javascript.js
+```
+
+#### Visual Regression testing with Cypress SDK
+
+Contributors can perform functional user interface and experience testing using [Cypress](https://www.cypress.io/). Cypress is not based on Selenium WebDriver and instead is built on JavaScript enabling significant performance advantages.
```

Comment:
This second sentence is irrelevant to a reader. If we use Cypress, we use Cypress. We should be telling people how to test, not trying to sell them on how Cypress is better than Selenium.